### DISCUSSION TITLE
Return a dict of claims instead of str?

### DISCUSSION THREAD

**Discussion by Aezore** [latchset/jwcrypto#313] [Q&A] [answered]
Not sure if I'm doing this correctly. JWT creation ask for a dict or a str for the claims attribute. Which I guess it either loads() the str or stores the dict directly. But when I create a jwt from a serialized token it only returns a str. Is there a way to return either one or the other as default? 

I'm new to this and I'm still toying with this stuff but in my head I would like to read the serialized token -> validate it's not expired or tampered and then return "sub" data to the client. 

So far in order to read the sub claim from the token is to loads(token.claims) and then read the "sub" key. Is this the intented procedure or am I missing something?

**Comment by simo5** [comment, marked as answer]
The jwk.claims storage is a json_encoded string.
You can use jwcrypto.common.json_decode(token.claims) to obtain a proper dict if that is the form you prefer to use.

**Reply by Aezore** [reply]
That's ok. I guess role separation it's a more maintainable approach (pass token.claims to another logic function to process instead of relying in your package). I was a bit confused as I was expecting to return a dict (since I supply a dict payload). Thanks for the quick response!
